examples/hnas/hpolib2_example.py

```diff
@@ -0,0 +1,113 @@
+"""
+This example show how to run AG on HPOBench (https://github.com/automl/HPOBench) benchmarks.
+For simplicity we use the NASBench201 benchmark.
+To run this benchmark you need to follow the installation guide of HPOBench.
+
+We run AG here on a single instance. If you want to run it across instances, have a look at the mlp_examply.py script
+in the example folder (that also shows you how to store intermediate results in a callback function).
+
+"""
+
+import autogluon.core as ag
+import time
+import logging
+import argparse
+
+from hpolib.benchmarks.nas.nasbench_201 import NasBench201BaseBenchmark
+
+
+def make_benchmark(dataset_name='cifar10-valid'):
+    b = NasBench201BaseBenchmark(dataset_name)
+    cs = b.get_configuration_space()
+
+    d = dict()
+    for h in cs.get_hyperparameters():
+        d[h.name] = ag.space.Categorical(*h.choices)
+
+    @ag.args(**d, epochs=199)
+    def objective_function(args, reporter, **kwargs):
+
+        ts_start = time.time()
+
+        config = dict()
+        for h in cs.get_hyperparameters():
+            config[h.name] = args[h.name]
+
+        for epoch in range(args.epochs):
+
+            res = b.objective_function(config, fidelity={'epoch': epoch})
+
+            acc = 1 - res['function_value'] / 100
+
+            ts_now = time.time()
+            eval_time = ts_start - ts_now
+            reporter(
+                epoch=epoch + 1,
+                performance=acc,
+                eval_time=eval_time,
+                time_step=ts_now, **config)
+
+    return objective_function
+
+
+# CLI
+def parse_args():
```

Comment:
At some point, we should centralize this stuff somewhere

Reply:
In fact, we might want to go back to the previous setup, where one could select the train_fn function.